yarn-project/prover-client/src/orchestrator/epoch-proving-state.ts

```diff
@@ -119,8 +121,8 @@ export class EpochProvingState {
       previousBlockHash,
       this,
     );
-    this.blocks.push(block);
-    if (this.blocks.length === this.totalNumBlocks) {
+    this.blocks[index] = block;
```

Comment:
I think CI is failing because there are holes in the array now and this function can't deal with that 😢 

https://github.com/AztecProtocol/aztec-packages/blob/cae1203ec4263d3b64fbc3fba5cfa281922004bd/yarn-project/prover-client/src/orchestrator/epoch-proving-state.ts#L178-L180

Reply:
Ahh good catch, thanks!

Reply:
Fixed in https://github.com/AztecProtocol/aztec-packages/pull/10350/commits/084c68a8a205cf408ef483ca7b5d705ff0b5e9f5